src/audio/pipeline/pipeline-graph.c

```diff
@@ -184,61 +184,11 @@ void pipeline_disconnect(struct comp_dev *comp, struct comp_buffer *buffer, int
 	irq_local_enable(flags);
 }
 
-static int pipeline_comp_free(struct comp_dev *current,
```

Comment:
Does this mean we have a double free today ? i.e. free by host then free by FW ?
I would prefer to keep parts of this so that FW can free anything the host does not free (and complain about it). e.g.
"hey you freed the pipeline but did not free component X in the pipeline, freeing X now"

Reply:
@lgirdwood we have ref counts for widget use on the host side. If the FW can free things without the host's knowledge that wouldnt work at all especially when widgets are shared between pipelines. I think it best to keep the freeing to just one side.

Reply:
also if you look at pipeline_new() and pipeline_free()today, they are not balanced in terms of what they do. WIth this change, pipeline_free() undoes exactly what pipeline_new() does